### DISCUSSION TITLE
Add support for the HTTP 103 Early Hints informational response

### DISCUSSION THREAD

**Discussion by user451421541757324** [encode/starlette#2800] [Ideas]
> The HTTP 103 Early Hints [informational response](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#informational_responses) may be sent by a server while it is still preparing a response, with hints about the sites and resources that the server expects the final response will link to. This allows a browser to [preconnect](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/preconnect) to sites or start [preloading](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/preload) resources even before the server has prepared and sent a final response. Preloaded resources indicated by early hints are fetched by the client as soon as the hints are received.

- [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/103)
- [Chrome for developers](https://developer.chrome.com/docs/web-platform/early-hints?hl=en)

Before/After [comparison of Early Hints](https://developer.chrome.com/static/docs/web-platform/early-hints/image/comparison-two-sites-157e25c61058d_1920.jpg) on a test website done with WebPageTest.

**Comment by Kludex** [comment]
Ideas for API? Is there any ASGI framework that implements this https://asgi.readthedocs.io/en/latest/extensions.html#early-hints ?

**Reply by user451421541757324** [reply]
> Ideas for API?

I was looking in the documentation at the Response section:

Signature: `Response(content, status_code=200, headers=None, media_type=None)`

What if:

Signature: `Response(content, status_code=200, headers=None, media_type=None, hints=None)`
Usage: `response = Response('Hello, world!', media_type='text/html', hints='Link: </style.css>; rel=preload; as=style')`

The [Bun folks](https://github.com/oven-sh/bun/issues/8690#issuecomment-1927372737) were thinking about using `yield`, but I don't know if it is applicable to Python.

> Is there any ASGI framework

A quick search found nothing.

Sorry if the answers seem trivial, I am a newbie.
